server/sonar-web/src/main/js/apps/projectActivity/components/StaticGraphs.js

```diff
@@ -0,0 +1,142 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import React from 'react';
+import moment from 'moment';
+import { some, sortBy } from 'lodash';
+import { AutoSizer } from 'react-virtualized';
+import AdvancedTimeline from '../../../components/charts/AdvancedTimeline';
+import StaticGraphsLegend from './StaticGraphsLegend';
+import { formatMeasure, getShortType } from '../../../helpers/measures';
+import { EVENT_TYPES_PRIORITY, generateCoveredLinesMetric } from '../utils';
+import { translate } from '../../../helpers/l10n';
+import type { Analysis, MeasureHistory } from '../types';
+
+type Props = {
+  analyses: Array<Analysis>,
+  eventFilter: string,
+  leakPeriodDate: Date,
+  loading: boolean,
+  measuresHistory: Array<MeasureHistory>,
+  metricsType: string,
+  seriesStyle?: { [string]: string }
+};
+
+export default class StaticGraphs extends React.PureComponent {
+  props: Props;
+
+  formatYTick = tick => formatMeasure(tick, getShortType(this.props.metricsType));
+
+  formatValue = value => formatMeasure(value, this.props.metricsType);
+
+  getEvents = () => {
+    const { analyses, eventFilter } = this.props;
+    const filteredEvents = analyses.reduce((acc, analysis) => {
+      if (analysis.events.length <= 0) {
+        return acc;
+      }
+      let event;
+      if (eventFilter) {
+        event = analysis.events.filter(event => event.category === eventFilter)[0];
+      } else {
+        event = sortBy(analysis.events, event => EVENT_TYPES_PRIORITY[event.category])[0];
+      }
+      if (!event) {
+        return acc;
+      }
+      return acc.concat({
+        className: event.category,
+        name: event.name,
+        date: moment(analysis.date).toDate()
+      });
+    }, []);
+    return sortBy(filteredEvents, 'date');
+  };
+
+  getSeries = () =>
+    sortBy(
+      this.props.measuresHistory.map((measure, idx) => {
+        if (measure.metric === 'uncovered_lines') {
+          return generateCoveredLinesMetric(measure, this.props.measuresHistory);
```

Comment:
to be discussed: we never ever show the number of **covered** lines, this will be the first place => inconsistency